connectors/src/connectors/google_drive/temporal/activities.ts

```diff
@@ -547,33 +542,47 @@ export async function incrementalSync(
       if (!change.file.id) {
         continue;
       }
-      if (!(await objectIsInFolder(oauth2client, change.file.id, foldersIds))) {
+
+      if (
+        !(await objectIsInFolders(
+          connectorId,
+          authCredentials,
+          await driveObjectToDustType(change.file, authCredentials),
+          selectedFoldersIds
+        ))
+      ) {
+        // The current file is not in the list of selected folders.
+        // If we have it locally, we need to garbage collect it.
+        const localFile = await GoogleDriveFiles.findOne({
+          where: {
+            connectorId: connectorId,
+            driveFileId: change.file.id,
+          },
+        });
+        if (localFile) {
+          await deleteOneFile(connectorId, change.file.id);
```

Comment:
In theory, we do not need to garbage collect as part of the incrementalSync workflow anymore since we have this code here now.
I tested moving a folder with files inside outside of the list of selected folders, and it works.

LMK what you think @spolu.